### PR TITLE
LPS-84629 When rdnEscape is false, we also need encode "\".

### PR DESCRIPTION
**Pull request by samuelkong**
cc @yxingwu

### CODE REVIEW

**Comment by samuelkong** [comment]
ci:test

File: modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/DefaultPortalLDAP.java [hunk 1]
```diff
@@ -89,11 +89,16 @@
 	@Override
 	public String encodeFilterAttribute(String attribute, boolean rdnEscape) {
 		String[] oldString = {
-			StringPool.STAR, StringPool.OPEN_PARENTHESIS,
+			StringPool.BACK_SLASH, StringPool.STAR, StringPool.OPEN_PARENTHESIS,
```
Comment (samuelkong) [inline modules/apps/portal-security/portal-security-ldap-impl/src/main/java/com/liferay/portal/security/ldap/internal/DefaultPortalLDAP.java]: Please note that `BACK_SLASH` needs to appear first in this list.

**Comment by samuelkong** [comment]
ci:retest

**Comment by topolik** [comment]
Pull request submitted at: https://github.com/brianchandotcom/liferay-portal/pull/62425
View total diff: https://github.com/topolik/liferay-portal/compare/4423c569ba...73aceca5ee

**Pull request closed without merging**